src/scripts/move.js

```diff
@@ -0,0 +1,257 @@
+  const emptyFields = [];
+
+  for (const cell of allCells) {
+    if (cell.classList.length === 1) {
+      emptyFields.push(cell);
+    }
+  }
+
+  // create new item:
+  if (emptyFields.length > 0) {
+    const randomIndex = randomField(0, emptyFields.length - 1);
+    const randomValue = randomField(0, values.length - 1);
+
+    emptyFields[randomIndex]
+      .classList.add('field-cell--' + values[randomValue]);
+
+    emptyFields[randomIndex]
+      .innerText = values[randomValue];
+
+  // if its impossible to create a new item:
+  } else {
+    messageAfterLose.classList.remove('hidden');
+    messageDuringTheGame.classList.add('hidden');
+
+    setNewHighestScore(gameScore, highestScore);
+
+    // mainListenerRemover();
+    document.body.removeEventListener('keyup', arrowsPlayHandler);
+  }
+}
+
+// Main movement algorithm:
+function arrowsPlayHandler(e) {
+  let isMoved = false;
+  const notEmptyFields = getNotEmptyFields(allCells);
+  const notEmptyFieldsPositions = getNotEmptyFieldsCoords(allCells);
+
+  switch (e.key) {
+    case 'ArrowDown':
+      for (const field of notEmptyFields.reverse()) {
+        const fieldPos = field.id.split('-').slice(1);
+
+        for (const cell of [...allCells].reverse()) {
+          const cellPos = cell.id.split('-').slice(1);
+
+          if (
+            checkForMove(field, cell, fieldPos, cellPos, 0, 1)
+          ) {
+            moveCell(field, cell, fieldPos, cellPos, 0);
+            isMoved = true;
+            break;
+          } else if (
+            checkForMerge(field, cell, fieldPos, cellPos, 0, 1)
+            && checkBetween(fieldPos, cellPos, notEmptyFieldsPositions, 0)
+          ) {
+            mergeCells(field, cell, fieldPos, cellPos, 0);
+            isMoved = true;
+            break;
+          }
+        }
+      }
+
+      if (isMoved || notEmptyFields.length === totalCountOfCells) {
+        setTimeout(() => {
+          createItemInRandomEmptyField();
+        }, 301);
+      }
+      break;
+
+    case 'ArrowUp':
+      for (const field of notEmptyFields) {
+        const fieldPos = field.id.split('-').slice(1);
+
+        for (const cell of [...allCells]) {
+          const cellPos = cell.id.split('-').slice(1);
+
+          if (
+            checkForMove(field, cell, fieldPos, cellPos, 0, 1, true)
+          ) {
+            moveCell(field, cell, fieldPos, cellPos, 0);
+            isMoved = true;
+            break;
+          } else if (
+            checkForMerge(field, cell, fieldPos, cellPos, 0, 1, true)
+            && checkBetween(fieldPos, cellPos, notEmptyFieldsPositions, 0, true)
+          ) {
+            mergeCells(field, cell, fieldPos, cellPos, 0);
+            isMoved = true;
+            break;
+          }
+        }
+      }
+
+      if (isMoved || notEmptyFields.length === totalCountOfCells) {
+        setTimeout(() => {
+          createItemInRandomEmptyField();
+        }, 301);
+      }
+      break;
```

Comment:
But you've used the same logic for every case, but with other direction values (correct me if I wrong)
You can create separate function for it and reuse with different direction values 🙂